docs/usage-guide/topics/ch11-resumption.md

```diff
@@ -25,6 +25,8 @@ Servers should set the three caching callback functions: `s2n_config_set_cache_s
 
 Clients should call `s2n_connection_get_session()` to retrieve some serialized state about the session. Then `s2n_connection_set_session()` should be called with that saved state when attempting to resume a new connection.
 
+Any errors during the connection will result in that specific session being removed from the cache, making it unavailable for resumption. This includes an abrupt connection termination where no `close_notify` alert is received by the server. Applications must perform a graceful TLS shutdown to be able to resume a session with session caching. For more information on how to properly close connections, see [Closing the Connection](./ch07-io.md#closing-the-connection)
+
```

Comment:
Couple comments:
* I wonder if it matters that the session isn't NECESSARILY "unavailable for resumption"? Like we just call the delete callback, the application defines what exactly that does. Which could mean that what's really missing from this documentation is a discussion of when each callback is triggered, not just a discussion of the delete callback.
* I'm not sure "abrupt connection termination" is clear enough. "Unexpected end-of-stream" or "Termination of the underlying stream" might be better.
* "where no close_notify alert is received by the server" also isn't clear. s2n-tls sends a close_notify to indicate most fatal errors, not just graceful shutdowns.

Reply:
I will add a note about when/what callback is called and what it could mean, but I have a few questions.

- Is our cache_delete_callback() just an example of what application might define? I want to confirm that this is not the default callback definition that applications get.
https://github.com/aws/s2n-tls/blob/ffe0bf42da8f139eff8fd2237f47fbde40b478fb/bin/common.c#L467


> s2n-tls sends a close_notify to indicate most fatal errors, not just graceful shutdowns.

- Does the client send close_notify in the case of unexpected end-of-stream? What I'm trying to say here is that when there is an unexpected end-of-stream, server does not receive close_notify, therefore server treats it as error, therefore session cache is deleted, therefore session resumption will fail. Could you explain what's not clear or potentially confusing?

Reply:
> Is our cache_delete_callback() just an example of what application might define?

Correct. Note that you linked to the bin directory-- it's just used by s2nc and s2nd, which are just for testing. You'll also see a note in the docs above where you edited that session caching isn't enabled unless the application implements all the callbacks.

> Does the client send close_notify in the case of unexpected end-of-stream?

The stream is closed so it can't send anything :) But my point is that "close_notify" doesn't necessarily indicate a graceful shutdown. It could indicate a fatal error instead (but you're right, not an S2N_ERR_CLOSED error). I'd probably just leave the bit about "close_notify" out and refer to the IO section on closing connections.

Reply:
That makes sense, I modified the doc to reflect your suggestion. I'm not sure what useful documentation I can provide for the other callbacks, so I'll limit the doc just for `cache_delete_callback()` for this PR. I still think it's worth mentioning the failed resumption behavior, since customers are likely to implement in a way the function name suggests. Let me know if I missed any of your suggestions